src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionClosePaymentRetryQueueConsumerTests.kt

```diff
@@ -241,6 +241,147 @@ class TransactionClosePaymentRetryQueueConsumerTests {
+      val authorizationCompletedEvent =
+        transactionAuthorizationCompletedEvent(AuthorizationResultDto.KO) as TransactionEvent<Any>
+      val closureErrorEvent = transactionClosureErrorEvent() as TransactionEvent<Any>
+
+      val events =
+        listOf(
+          activationEvent,
+          authorizationRequestEvent,
+          authorizationCompletedEvent,
+          closureErrorEvent)
+
+      val expectedUpdatedTransaction =
+        transactionDocument(
+          TransactionStatusDto.UNAUTHORIZED, ZonedDateTime.parse(activationEvent.creationDate))
+
+      val transactionDocument =
+        transactionDocument(
+          TransactionStatusDto.CLOSURE_ERROR, ZonedDateTime.parse(activationEvent.creationDate))
+
+      /* preconditions */
+      given(checkpointer.success()).willReturn(Mono.empty())
+      given(
+          transactionsEventStoreRepository.findByTransactionIdOrderByCreationDateAsc(
+            TRANSACTION_ID))
+        .willReturn(events.toFlux())
+      given(transactionsViewRepository.findByTransactionId(TRANSACTION_ID))
+        .willReturn(Mono.just(transactionDocument))
+      given(transactionsViewRepository.save(viewArgumentCaptor.capture())).willAnswer {
+        Mono.just(it.arguments[0])
+      }
+      given(transactionClosedEventRepository.save(closedEventStoreRepositoryCaptor.capture()))
+        .willAnswer { Mono.just(it.arguments[0]) }
+      given(
+          nodeService.closePayment(
+            TransactionId(TRANSACTION_ID), ClosePaymentRequestV2Dto.OutcomeEnum.KO))
+        .willReturn(
+          ClosePaymentResponseDto().apply { outcome = ClosePaymentResponseDto.OutcomeEnum.KO })
+
+      /* test */
+
+      StepVerifier.create(
+          transactionClosureErrorEventsConsumer.messageReceiver(
+            BinaryData.fromObject(closureErrorEvent).toBytes(), checkpointer))
+        .expectNext()
+        .verifyComplete()
+
+      /* Asserts */
+      verify(checkpointer, Mockito.times(1)).success()
+      verify(nodeService, Mockito.times(1))
+        .closePayment(TransactionId(TRANSACTION_ID), ClosePaymentRequestV2Dto.OutcomeEnum.KO)
+      verify(transactionClosedEventRepository, Mockito.times(1))
+        .save(
+          any()) // FIXME: Unable to use better argument captor because of misbehaviour in static
+      // mocking
+      verify(transactionsViewRepository, Mockito.times(1)).save(expectedUpdatedTransaction)
+      verify(paymentGatewayClient, times(0)).requestVPosRefund(any())
+      verify(closureRetryService, times(0)).enqueueRetryEvent(any(), any())
+      assertEquals(TransactionStatusDto.UNAUTHORIZED, viewArgumentCaptor.value.status)
+      assertEquals(
+        TransactionEventCode.TRANSACTION_CLOSURE_FAILED_EVENT,
+        closedEventStoreRepositoryCaptor.value.eventCode)
+      assertEquals(
+        TransactionClosureData.Outcome.KO,
+        closedEventStoreRepositoryCaptor.value.data.responseOutcome)
+    }
+
```

Comment:
Add test for legacy TransactionClosureRetriedEvent event too

Reply:
Fixed